siege_utilities/schema/tests/test_migration_runner_integration.py

```
@@ -0,0 +1,213 @@
+    _write(migrations_dir, "V0001__create_thing.sql", textwrap.dedent("""\
+        CREATE SCHEMA IF NOT EXISTS sut_target;
+        CREATE TABLE sut_target.thing (id INT PRIMARY KEY);
+        """))
+    runner = _runner(pg_dsn, migrations_dir, tracking_schema)
+
+    applied = runner.apply_all()
+
+    assert [m.version for m in applied] == ["0001"]
+    # Tracking table exists and has the row.
+    with psycopg.connect(pg_dsn) as conn, conn.cursor() as cur:
+        cur.execute(f"SELECT version FROM {tracking_schema}._schema_migrations ORDER BY version")
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)